src/widgets/table.rs

```diff
@@ -173,207 +299,164 @@ pub enum HighlightSpacing {
+/// [`selected`]: TableState::selected()
+///
+/// See the [table example] and the recipe and traceroute tabs in the [demo2 example] for a more in
+/// depth example of the various configuration options and for how to handle state.
+///
+/// [table example]: https://github.com/ratatui-org/ratatui/blob/master/examples/table.rs
+/// [demo2 example]: https://github.com/ratatui-org/ratatui/blob/master/examples/demo2/
+///
+/// # Example
 ///
-/// It is a collection of [`Row`]s, themselves composed of [`Cell`]s:
 /// ```rust
-/// use ratatui::{prelude::*, widgets::*};
+/// # use ratatui::{prelude::*, widgets::*};
+/// # fn ui(frame: &mut Frame) {
+/// # let area = Rect::default();
+/// # let rows = vec![Row::new(vec!["Cell1", "Cell2"])];
+/// # let widths = vec![];
+/// // This should be stored outside of the function in your application state.
+/// let table = Table::new(rows).widths(widths);
 ///
-/// Table::new(vec![
-///     // Row can be created from simple strings.
-///     Row::new(vec!["Row11", "Row12", "Row13"]),
-///     // You can style the entire row.
-///     Row::new(vec!["Row21", "Row22", "Row23"]).style(Style::default().fg(Color::Blue)),
-///     // If you need more control over the styling you may need to create Cells directly
-///     Row::new(vec![
-///         Cell::from("Row31"),
-///         Cell::from("Row32").style(Style::default().fg(Color::Yellow)),
-///         Cell::from(Line::from(vec![
-///             Span::raw("Row"),
-///             Span::styled("33", Style::default().fg(Color::Green))
-///         ])),
-///     ]),
-///     // If a Row need to display some content over multiple lines, you just have to change
-///     // its height.
-///     Row::new(vec![
-///         Cell::from("Row\n41"),
-///         Cell::from("Row\n42"),
-///         Cell::from("Row\n43"),
-///     ]).height(2),
-/// ])
-/// // You can set the style of the entire Table.
-/// .style(Style::default().fg(Color::White))
-/// // It has an optional header, which is simply a Row always visible at the top.
-/// .header(
-///     Row::new(vec!["Col1", "Col2", "Col3"])
-///         .style(Style::default().fg(Color::Yellow))
-///         // If you want some space between the header and the rest of the rows, you can always
-///         // specify some margin at the bottom.
-///         .bottom_margin(1)
-/// )
-/// // As any other widget, a Table can be wrapped in a Block.
-/// .block(Block::default().title("Table"))
-/// // Columns widths are constrained in the same way as Layout...
-/// .widths(&[Constraint::Length(5), Constraint::Length(5), Constraint::Length(10)])
-/// // ...and they can be separated by a fixed spacing.
-/// .column_spacing(1)
-/// // If you wish to highlight a row in any specific way when it is selected...
-/// .highlight_style(Style::default().add_modifier(Modifier::BOLD))
-/// // ...and potentially show a symbol in front of the selection.
-/// .highlight_symbol(">>");
+/// let mut table_state = TableState::default();
```

Comment:
Looking at this again because I'm documenting list, I think this is the state that should be stored in the app state, not the widget.

Reply:
Arh line selection took the entire diff ...
Anyway my comment just references the geen lines

Reply:
got it - fixed